presto-orc/src/main/java/com/facebook/presto/orc/writer/LongDictionaryColumnWriter.java

```diff
@@ -0,0 +1,300 @@
+        for (int i = 0; i < dictionary.size(); i++) {
+            this.dictionaryDataStream.writeLong(elements[i]);
+        }
+
+        columnEncoding = new ColumnEncoding(DICTIONARY, dictionary.size());
+        return null;
+    }
+
+    @Override
+    protected void writePresentAndDataStreams(
+            int rowGroupValueCount,
+            IntBigArray rowGroupIndexes,
+            int[] originalDictionaryToSortedIndex,
+            PresentOutputStream presentStream,
+            LongOutputStream dataStream)
+    {
+        checkArgument(originalDictionaryToSortedIndex == null, "Unsupported originalDictionaryToSortedIndex");
+        for (int position = 0; position < rowGroupValueCount; position++) {
+            presentStream.writeBoolean(rowGroupIndexes.get(position) != UNUSED_VALUE);
+        }
+        for (int position = 0; position < rowGroupValueCount; position++) {
+            int index = rowGroupIndexes.get(position);
+            // index == UNUSED_VALUE represents null in the dictionary.
+            if (index != UNUSED_VALUE) {
+                dataStream.writeLong(index);
+            }
+        }
+    }
+
+    @Override
+    protected List<StreamDataOutput> getDictionaryStreams(int column)
+    {
+        return asList(dictionaryDataStream.getStreamDataOutput(column));
+    }
+
+    @Override
+    protected ColumnStatistics createColumnStatistics()
+    {
+        ColumnStatistics statistics = statisticsBuilder.buildColumnStatistics();
+        statisticsBuilder = new IntegerStatisticsBuilder();
+        return statistics;
+    }
+
+    @Override
+    protected void closeDictionary()
+    {
+        dictionary = null;
+        dictionaryDataStream.close();
+    }
+
+    @Override
+    protected void resetDictionary()
+    {
+        columnEncoding = null;
+        dictionary = new LongDictionary(10_000);
+        dictionaryDataStream.reset();
+        statisticsBuilder = new IntegerStatisticsBuilder();
+        directValues = null;
+        directNulls = null;
+    }
+
+    private static class Long2IntMapWithByteSize
+            extends Long2IntOpenHashMap
+    {
+        public Long2IntMapWithByteSize(int expectedEntries, int unusedValue)
+        {
+            super(expectedEntries);
+            this.defaultReturnValue(unusedValue);
+        }
+        public int getRetainedBytes()
+        {
+            return this.key.length * 8 + this.value.length * 4;
+        }
+    }
+
+    private static class LongDictionary
+    {
+        private final Long2IntMapWithByteSize dictionary;
+        private final LongArrayList elements;
+
+        public LongDictionary(int expectedEntries)
+        {
+            this.dictionary = new Long2IntMapWithByteSize(expectedEntries, UNUSED_VALUE);
+            this.elements = new LongArrayList(expectedEntries);
+        }
+
+        public long getRetainedBytes()
+        {
+            return this.dictionary.getRetainedBytes() + sizeOf(this.elements.elements());
```

Comment:
Remove `this.`. We don't use `this.` outside constructors. Same for other functions